app/src/org/commcare/models/database/SqlStorage.java

```diff
@@ -94,16 +103,23 @@ public SqlStorage(String table, Class<? extends T> ctype, AndroidDbHelper helper
         }
 
         Cursor c = db.query(table, new String[]{DatabaseHelper.ID_COL}, whereClause.first, whereClause.second, null, null, null);
-        return fillIdWindow(c, DatabaseHelper.ID_COL);
+        return fillIdWindow(c, DatabaseHelper.ID_COL, returnSet);
     }
 
     public static Vector<Integer> fillIdWindow(Cursor c, String columnName) {
+        return fillIdWindow(c, columnName, null);
+    }
+
+    public static Vector<Integer> fillIdWindow(Cursor c, String columnName, LinkedHashSet<Integer> newReturn) {
```

Comment:
Why the use of LinkedHashSet and then this pass-in-returned-object pattern (is there a real name for that?) Seems redundant with the method's return value